### PR TITLE
Add TCK tests for registering custom scopes/contexts with Build Compatible Extensions

### LINKED ISSUE #404: Lite TCK should have a test for custom CDI context registration
I didn't find any test for custom context registered via `BuildCompatibleExtension`.
[There is a test for similar functionality for CDI Full ](https://github.com/jakartaee/cdi-tck/tree/master/impl/src/main/java/org/jboss/cdi/tck/tests/full/context/alterable)(and classic `PortableExtension`) - we could use the very same thing and register it via BCE.

### PR DESCRIPTION
**Pull request by Ladicek**
Fixes #404

### CODE REVIEW

File: impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package org.jboss.cdi.tck.tests.build.compatible.extensions.customPseudoScope;
+
+import java.util.UUID;
+
+@Prototype
```
Comment (manovotn) [inline impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java:5]: This isn't bean defining annotation, you'll need to add a stereotype to this class or have a stereotype that already contains `@Prototype`.
See also `org.jboss.cdi.tck.tests.full.deployment.discovery.BeanDiscoveryTest` and its `MyStereotype` class which does the exact same to register custom pseudo scope.

**Review by manovotn — CHANGES_REQUESTED** [review]

File: impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customNormalScope/CustomNormalScopeExtension.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.enterprise.inject.build.compatible.spi.MetaAnnotations;
+import jakarta.enterprise.inject.build.compatible.spi.ScannedClasses;
+import jakarta.enterprise.inject.build.compatible.spi.Synthesis;
+import jakarta.enterprise.inject.build.compatible.spi.SyntheticComponents;
+
+public class CustomNormalScopeExtension implements BuildCompatibleExtension {
+    @Discovery
+    public void discovery(MetaAnnotations meta, ScannedClasses scan) {
+        meta.addContext(CommandScoped.class, CommandContext.class);
+        scan.add(CommandExecutor.class.getName());
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customNormalScope/CustomNormalScopeExtension.java:15]: Ah good point, this is actually inside a bean archive. Thanks, will remove.

File: impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java (hunk 1 above)
Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java:5]: I forgot that pseudo-scope annotations are not BDAs :-/ Will add a stereotype.

File: impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customNormalScope/CustomNormalScopeExtension.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customNormalScope/CustomNormalScopeExtension.java]: Done.

File: impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java (hunk 1 above)
Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/customPseudoScope/PrototypeBean.java]: Done.

**Review by manovotn — APPROVED** [review]
Looks good now!

**Pull request merged**